easybuild/framework/easyconfig/tweak.py

```diff
@@ -324,13 +324,25 @@ def __repr__(self):
                     _log.debug("Overwriting %s with %s" % (key, fval))
                 ectxt = regexp.sub("%s = %s" % (res.group('key'), newval), ectxt)
                 _log.info("Tweaked %s list to '%s'" % (key, newval))
-            elif get_easyconfig_parameter_default(key) != val:
+            elif not is_easyconfig_parameter_default_value(key, val):
                 additions.append("%s = %s" % (key, val))
 
             tweaks.pop(key)
 
     # add parameters or replace existing ones
+    special_values = ('True', 'False', 'None')
+    quoted_special_values = ['"%s"' % val for val in special_values] + ["'%s'" % val for val in special_values]
     for (key, val) in tweaks.items():
+        # if the value is True/False/None then take that
+        # if e.g. (literal) True is wanted, then it can be passed as "True"/'True'
+        if val in special_values:
+            str_val = val
+            val = eval(val)
+        elif val in quoted_special_values:
+            str_val = val
+            val = val.strip('"\'')
+        else:
+            str_val = quote_str(val)
```

Comment:
this can be simplified a bit:
```suggestion
    special_values = {
        'True': True, "'True'": 'True', '"True"': 'True',
        'False': False, "'False'": 'False', '"False"': 'False',
        'None': None, "'None'": 'None', '"None"': 'None',
    }
    for (key, val) in tweaks.items():
        # if the value is True/False/None then take that
        # if e.g. (literal) True is wanted, then it can be passed as "True"/'True'
        if val in special_values:
            str_val = val
            val = special_values[val]
        else:
            str_val = quote_str(val)
```